openfeature/api.py

```diff
@@ -67,3 +81,30 @@ def get_hooks() -> typing.List[Hook]:
 
 def shutdown() -> None:
     _provider_registry.shutdown()
+
+
+def on(event: ProviderEvent, handler: EventHandler) -> None:
+    _event_support.add_global_handler(event, handler)
+
+
+def _register_client_handler(
+    client: OpenFeatureClient, event: ProviderEvent, handler: EventHandler
+) -> None:
+    _clients_with_handlers.add(client)
+    _event_support.add_client_handler(client, event, handler)
+
+
+def _run_handlers_for_provider(
+    provider: FeatureProvider,
+    event: ProviderEvent,
+    provider_details: ProviderEventDetails,
+) -> None:
+    details = EventDetails.from_provider_event_details(
+        provider.get_metadata().name, provider_details
+    )
+    # run the global handlers
+    _event_support.run_global_handlers(event, details)
+    # run the handlers for clients associated to this provider
+    for client in _clients_with_handlers:
+        if client.provider == provider:
+            _event_support.run_client_handlers(client, event, details)
```

Comment:
This is the key dispatching logic to review, not sure I got it right, might need further tweaking.